src/WxWork/Provider.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace SocialiteProviders\WeixinWork;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Str;
+use SocialiteProviders\Manager\Contracts\ConfigInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'WEIXINWORK';
+
+    protected $scopes = ['snsapi_base'];
+
+    protected $userid;
+
+    /**
+     * @param \SocialiteProviders\Manager\Contracts\OAuth1\ProviderInterface|\SocialiteProviders\Manager\Contracts\OAuth2\ProviderInterface $config
+     */
+    public function setConfig(ConfigInterface $config)
+    {
+        $config = config('services.weixinwork');
+
+        $this->config = $config;
+        $this->corpid = $config['corpid'];
+        $this->clientId = $config['client_id'];
+        $this->clientSecret = $config['client_secret'];
+        $this->redirectUrl = $config['redirect'];
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function getAuthUrl($state)
+    {
+        $isWxWork = Str::is('*wxwork*', request()->userAgent());
```

Comment:
I assume this is detecting the mobile app?

Reply:
Used to check if it is running inside wxwork.